src/backend/tests/integration/test_openai_streaming_comparison.py

```diff
@@ -0,0 +1,343 @@
+    if response.status_code != 201:
+        logger.error(f"Failed to create global variable: {response.content}")
+        return False
+
+    logger.info(f"Successfully created global variable: {name}")
+    return True
+
+
+async def load_and_prepare_flow(client: AsyncClient, created_api_key):
+    """Load Simple Agent flow and wait for it to be ready."""
+    headers = {"x-api-key": created_api_key.api_key}
+
+    # Create OPENAI_API_KEY global variable
+    openai_api_key = os.getenv("OPENAI_API_KEY")
+    if not openai_api_key:
+        pytest.skip("OPENAI_API_KEY environment variable not set")
+
+    await create_global_variable(client, headers, "OPENAI_API_KEY", openai_api_key)
+
+    # Load the Simple Agent template
+    template_path = (
+        pathlib.Path(__file__).resolve().parent.parent.parent
+        / "base"
+        / "langflow"
+        / "initial_setup"
+        / "starter_projects"
+        / "Simple Agent.json"
+    )
+
+    flow_data = await asyncio.to_thread(lambda: json.loads(pathlib.Path(template_path).read_text()))
+
+    # Add the flow
+    response = await client.post("/api/v1/flows/", json=flow_data, headers=headers)
+    assert response.status_code == 201
+    flow = response.json()
+
+    # Poll for flow builds to complete
+    max_attempts = 10
+    for attempt in range(max_attempts):
+        builds_response = await client.get(f"/api/v1/monitor/builds?flow_id={flow['id']}", headers=headers)
+
+        if builds_response.status_code == 200:
+            builds = builds_response.json().get("vertex_builds", {})
+            all_valid = True
+            for build_list in builds.values():
+                if not build_list or build_list[0].get("valid") is not True:
+                    all_valid = False
+                    break
+
+            if all_valid and builds:
+                break
+
+        if attempt < max_attempts - 1:
+            await asyncio.sleep(1)
+
+    return flow, headers
+
+
+@pytest.mark.api_key_required
+@pytest.mark.integration
+async def test_openai_streaming_format_comparison(client: AsyncClient, created_api_key):
+    """Compare raw HTTP streaming formats between OpenAI and our API."""
+    # Test input
+    input_msg = "What is 25 + 17? Use your calculator tool."
+
+    # Tools definition
+    tools = [
+        {
+            "type": "function",
+            "name": "evaluate_expression",
+            "description": "Perform basic arithmetic operations on a given expression.",
+            "parameters": {
+                "type": "object",
+                "properties": {
+                    "expression": {
+                        "type": "string",
+                        "description": "The arithmetic expression to evaluate (e.g., '4*4*(33/22)+12-20').",
+                    }
+                },
+                "required": ["expression"],
+            },
+        }
+    ]
+
+    # Get OpenAI API key
+    openai_api_key = os.getenv("OPENAI_API_KEY")
+    if not openai_api_key:
```

Comment:
You may need to check if it isn't a dummy api key here. Github will do that sometimes